staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -92,7 +92,7 @@ func (o *RecommendedOptions) ApplyTo(config *server.RecommendedConfig, scheme *r
 	if err := o.Authorization.ApplyTo(&config.Config.Authorization); err != nil {
 		return err
 	}
-	if err := o.Audit.ApplyTo(&config.Config); err != nil {
+	if err := o.Audit.ApplyTo(&config.Config, config.ClientConfig, config.SharedInformerFactory, DefaultProcessInfo(), nil, nil); err != nil {
```

Comment:
I do not think the generic apiserver should default to "kube-apiserver"

Reply:
not sure what it should default to, just `apiserver` for name, and `kube-system` for namespace? cc @deads2k @sttts

Reply:
> not sure what it should default to, just apiserver for name, and kube-system for namespace

I wouldn't make either of those assumptions... if you require info about the component, should we add parameters when constructing RecommendedOptions?

Reply:
yea I had initially done that but @deads2k said to try and keep it out of there, I'm not sure how else we would get those values though

Reply:
ok updated with adding to the recommended options, this leaves the question of whether the auth resolver and service resolver should be setup that way as well?